### PR TITLE
Feature/setup dependency injection

### PR DESCRIPTION
**Pull request by MikeAde1**

### CODE REVIEW

File: app/src/main/java/com/ceracare/careplan/App.kt [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
 import android.app.Application
+import androidx.appcompat.app.AppCompatDelegate
+import dagger.hilt.android.HiltAndroidApp
 
+@HiltAndroidApp
 class App : Application() {
-
+    override fun onCreate() {
+        super.onCreate()
+        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
```
Comment (segunfrancis) [inline app/src/main/java/com/ceracare/careplan/App.kt:11]: We could just delete the theme file for night mode and also change the app theme to `Theme.MaterialComponents.Light.DarkActionBar`

File: app/src/main/java/com/ceracare/careplan/di/NetworkModule.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    fun provideApiInterface(retrofit: Retrofit): ApiInterface {
+        return retrofit.create(ApiInterface::class.java)
+    }
+}
+
+private fun provideOkHttpClient(): OkHttpClient {
+    val b = OkHttpClient.Builder()
+    b.connectTimeout(60, TimeUnit.SECONDS)
+    b.readTimeout(60, TimeUnit.SECONDS)
+    b.writeTimeout(60, TimeUnit.SECONDS)
```
Comment (segunfrancis) [inline app/src/main/java/com/ceracare/careplan/di/NetworkModule.kt:39]: we will later move the constants to a separate file. 
I personally feel 1 minute is too much for a network timeout. The client's are most likely based in Europe and they generally have a fast internet connection. 20-30 seconds will be better

**Review by segunfrancis — APPROVED** [review]
Nice work, you are really good at this and I am going to be learning a lot from you

File: app/src/main/java/com/ceracare/careplan/di/NetworkModule.kt (hunk 2 above)
Comment (MikeAde1) [reply app/src/main/java/com/ceracare/careplan/di/NetworkModule.kt:39]: Alright

**Pull request merged**